src/Symfony/Bundle/FrameworkBundle/Console/Application.php

```diff
@@ -28,16 +28,19 @@
 class Application extends BaseApplication
 {
     private $kernel;
+    private $loadDefaultCommands;
     private $commandsRegistered = false;
 
     /**
      * Constructor.
      *
-     * @param KernelInterface $kernel A KernelInterface instance
+     * @param KernelInterface $kernel              A KernelInterface instance
+     * @param bool            $loadDefaultCommands Load all the default commands
      */
-    public function __construct(KernelInterface $kernel)
+    public function __construct(KernelInterface $kernel, $loadDefaultCommands = true)
     {
         $this->kernel = $kernel;
+        $this->loadDefaultCommands = $loadDefaultCommands;
```

Comment:
Technically you could pre-set `$this->commandsRegistered` here.. we dont really need the extra property imo.